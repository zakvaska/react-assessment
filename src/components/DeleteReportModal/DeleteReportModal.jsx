import React, { useEffect, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import Modal from '../Modal/Modal';
import {
  ModalBody,
  ModalHeader,
  ModalFooter,
  ButtonList,
  StyledButton,
  StyledSelect,
  TextInput
} from '../styled-components/Task';

const DeleteReportModal = ({ modalCloseHandler, onConfirm, onCancel }) => {
  const [textInputValue, setTextInputValue] = useState('');
  const [selectedReport, setSelectedReport] = useState([]);
  const [canDelete, setCanDelete] = useState(false);

  const changeSelection = (e) => {
    setSelectedReport(e.target.value);
  };

  const changeValue = (e) => {
    setTextInputValue(e.target.value);
  };

  const handleDeletion = () => {
    console.log(`Rerort number ${selectedReport} will be deleted`);
    onConfirm(selectedReport);
  };

  useEffect(() => {
    if (textInputValue === 'Delete' && !Array.isArray(selectedReport)) {
      setCanDelete(true);
    } else {
      setCanDelete(false);
    }
  }, [textInputValue, selectedReport]);

  return (
    <Modal modalCloseHandler={modalCloseHandler}>
      <ModalHeader>
        <FontAwesomeIcon icon={faExclamationTriangle} color="#d0011b" />
        <div>Are you sure you want to delete this report and its history?</div>
      </ModalHeader>
      <ModalBody>
        If you delete the&nbsp;
        <strong>Executive metrics</strong>
        &nbsp;report, you will also delete the associated history:
        <StyledSelect multiple onChange={changeSelection}>
          <option value="1">January 2020</option>
          <option value="2">February 2020</option>
          <option value="3">March 2020</option>
          <option value="4">April 2020</option>
          <option value="5">May 2020</option>
          <option value="6">June 2020</option>
          <option value="7">Jule 2020</option>
          <option value="8">August 2020</option>
          <option value="9">September 2020</option>
          <option value="10">October 2020</option>
          <option value="11">November 2020</option>
          <option value="12">December 2020</option>
        </StyledSelect>
        Please type the word &apos;Delete&apos; to remove the&nbsp;
        <strong>Executive metrics</strong>
        &nbsp;report and its associated history:
        <TextInput type="text" onChange={changeValue} />
      </ModalBody>
      <ModalFooter>
        <ButtonList justify="flex-end" marginRight="1rem">
          <StyledButton type="button" onClick={handleDeletion} disabled={!canDelete}>Delete all</StyledButton>
          <StyledButton type="button" onClick={onCancel}>Cancel</StyledButton>
        </ButtonList>
      </ModalFooter>
    </Modal>
  );
};

export default DeleteReportModal;
