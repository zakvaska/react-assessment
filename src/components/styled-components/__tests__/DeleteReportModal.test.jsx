import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import DeleteReportModal from '../../DeleteReportModal/DeleteReportModal';

it('DeleteReportsModal renders correctly', () => {
  const handleClose = jest.fn();
  const handleConfirm = jest.fn();
  const handleCancel = jest.fn();
  const { container } = render(
    <DeleteReportModal 
      modalCloseHandler={handleClose}
      onConfirm={handleConfirm}
      onCancel={handleCancel}  
    />
  );
  expect(container).toMatchSnapshot();
});
