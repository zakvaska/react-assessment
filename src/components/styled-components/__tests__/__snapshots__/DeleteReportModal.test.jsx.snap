// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteReportsModal renders correctly 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 > button:not(:last-of-type) {
  margin-right: 1rem;
}

.c7 > button {
  min-width: 7rem;
}

.c8 {
  color: #24619a;
  background: #e8f4ff;
  border: 1px solid transparent;
  padding: 0.7rem 1rem;
  border-radius: 0.3rem;
  margin-top: 2rem;
  cursor: pointer;
  font-family: sans-serif;
  font-size: 1rem;
  font-weight: bold;
  -webkit-transition: background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
}

.c8:hover:not(:disabled) {
  background: white;
  border: 1px solid #24619a;
}

.c8:disabled {
  color: #24619a73;
  cursor: default;
}

.c0 {
  position: absolute;
  width: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  background-color: white;
  border: 1px solid #d10720;
  width: 35rem;
  border-radius: .75rem;
  margin: 10rem auto;
  padding: 1.5rem;
  color: black;
  text-align: start;
  font-family: sans-serif;
  font-size: 1.1rem;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #e1e1e1;
}

.c2 > div {
  padding-left: .5rem;
  font-weight: bold;
  color: black;
  font-family: sans-serif;
}

.c3 {
  font-family: sans-serif;
  font-size: 1rem;
  padding-top: 1rem;
}

.c6 {
  padding-top: 1rem;
}

.c4 {
  display: block;
  width: 100%;
  height: 7rem;
  margin-top: 3rem;
  margin-bottom: 1rem;
  padding: .5rem;
  font-size: 1rem;
  color: #3970a3;
  border-color: #9d9d9d;
}

.c4 > option {
  font-weight: bold;
}

.c5 {
  display: block;
  margin-top: 1rem;
  padding-left: .5rem;
  border: 1px solid #f3f3f3;
  line-height: 1.5rem;
  box-shadow: 0 1px 1px rgb(0 0 0 / 20%);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
          color="#d0011b"
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
        <div>
          Are you sure you want to delete this report and its history?
        </div>
      </div>
      <div
        class="c3"
      >
        If you delete the 
        <strong>
          Executive metrics
        </strong>
         report, you will also delete the associated history:
        <select
          class="c4"
          multiple=""
        >
          <option
            value="1"
          >
            January 2020
          </option>
          <option
            value="2"
          >
            February 2020
          </option>
          <option
            value="3"
          >
            March 2020
          </option>
          <option
            value="4"
          >
            April 2020
          </option>
          <option
            value="5"
          >
            May 2020
          </option>
          <option
            value="6"
          >
            June 2020
          </option>
          <option
            value="7"
          >
            Jule 2020
          </option>
          <option
            value="8"
          >
            August 2020
          </option>
          <option
            value="9"
          >
            September 2020
          </option>
          <option
            value="10"
          >
            October 2020
          </option>
          <option
            value="11"
          >
            November 2020
          </option>
          <option
            value="12"
          >
            December 2020
          </option>
        </select>
        Please type the word 'Delete' to remove the 
        <strong>
          Executive metrics
        </strong>
         report and its associated history:
        <input
          class="c5"
          type="text"
        />
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <button
            class="c8"
            disabled=""
            type="button"
          >
            Delete all
          </button>
          <button
            class="c8"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
.c0 {
  max-height: 300px;
  overflow: scroll;
  margin-top: 15px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
